components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementService.java

```diff
@@ -166,12 +170,27 @@ private RegistrationResponseProfile createOAuthApplication(RegistrationRequestPr
 
             OAuthConsumerAppDTO oAuthConsumerApp = new OAuthConsumerAppDTO();
             oAuthConsumerApp.setApplicationName(applicationName);
+
             //TODO: After implement multi-urls to the oAuth application, we have to change this API call
-            if (profile.getRedirectUris().size() == 0) {
+            if (profile.getRedirectUris().size() == 0 && (profile.getGrantTypes().contains(
+                    DCRConstants.GrantTypes.AUTHORIZATION_CODE) || profile.getGrantTypes().
+                    contains(DCRConstants.GrantTypes.IMPLICIT))) {
                 String errorMessage = "RedirectUris property must have at least one URI value.";
-                throw IdentityException.error(DCRException.class, ErrorCodes.META_DATA_VALIDATION_FAILED.toString(), errorMessage);
+                throw IdentityException.error(DCRException.class, ErrorCodes.META_DATA_VALIDATION_FAILED.toString(),
+                        errorMessage);
+            } else if (profile.getRedirectUris().size() == 1) {
+                String redirectUri = profile.getRedirectUris().get(0);
+                if (DCRUtils.isRedirectionUriValid(redirectUri)) {
+                    oAuthConsumerApp.setCallbackUrl(redirectUri);
+                } else {
+                    //TODO: need to add error code
+                    throw IdentityException.error(DCRException.class, "Redirect URI " + redirectUri + " is invalid");
```

Comment:
Better to change the log messages to have a delimiter around the parameters. The printed message would look funny sometimes, without proper delimiter.
e.g. "Redirect URI " + redirectUri + " is invalid" -->"Redirect URI: " + redirectUri + ", is invalid"